### PR TITLE
Add revisionHistoryLimit to all deployments

### LINKED ISSUE #25058: Add revisionHistoryLimit to Helm deployments
### Description

We can use this parameter to control how many old ReplicaSets we want to retain for each deployment 

### Use case/motivation

We wouldn't end up with a lot of unused replicasets in our deployment

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by csp33**
closes: #25058

We can use this parameter to control how many old ReplicaSets we want to retain for each deployment.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Makes sense - but should not it be possible to define also "global" revisionHistoryLimit that would override all "detailed" ones if not set? That would be easier to control @jedcunningham @ephraimbuddy @dstandish ?

**Comment by csp98** [comment]
yeah, that makes sense. I will update this PR with that global value

**Comment by csp98** [comment]
Done @potiuk 
We can now specify _revisionHistoryLimit_ both globally and inside every component. The value from each component has priority over the global one.

**Review by potiuk — APPROVED** [review]
LGTM. Anyone wants to take a look @jedcunningham @dstandish @ephraimbuddy ?

**Comment by jedcunningham** [comment]
@csp98, can you also add corresponding sections to `values.yaml`, e.g.

```
workers:
  replicas: 1
  revisionHistoryLimit: ~
```

You'll also have to add `null` to the allowed types, but I think most folks just look at values.yaml to see possible options so it's worth adding there as well.

**Review by jedcunningham — APPROVED** [review]

File: tests/charts/test_statsd.py [hunk 1]
```diff
@@ -51,6 +52,17 @@ def test_should_add_volume_and_volume_mount_when_exist_extra_mappings(self):
             "subPath": "mappings.yml",
         } in jmespath.search("spec.template.spec.containers[0].volumeMounts", docs[0])
 
+    @parameterized.expand([(8,), (None,)])
```
Comment (jedcunningham) [inline tests/charts/test_statsd.py]: Why isn't statsd also getting coverage for the global setting?

Comment (csp33) [reply tests/charts/test_statsd.py]: My fault! Fixed

**Review by potiuk — APPROVED** [review]
Much nicer now.

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
I've restarted the failures, hopefully they pass this time around.

**Comment by csp98** [comment]
Cool @jedcunningham, thanks.
I also noticed that the branch is outdated (again 😅). Shall I rebase it when tests finish?

**Comment by jedcunningham** [comment]
No, you'll drive yourself crazy trying to keep up 👍. As long as it stays relatively up to date we are okay, plus we don't get a ton of chart changes anyways.

**Comment by potiuk** [comment]
> No, you'll drive yourself crazy trying to keep up +1. As long as it stays relatively up to date we are okay, plus we don't get a ton of chart changes anyways.

Well. Actually rebasing PR is a daily routine I do. The more often you do it, the less painful it is, and the easier it is. And you learn how to rebase quickly even if you have conflicts. This might sound counter-intiuitive but it actually works.

So contrary to what @jedcunningham said - yeah, by all means, rebase early, rebase often.

**Comment by potiuk** [comment]
:stuck_out_tongue:

**Review by potiuk — APPROVED** [review]

**Comment by jedcunningham** [comment]
Daily, sure, but it was already rebased 3 hours ago 🙃

Thanks @csp98! Congrats on your first commit

**Comment by csp98** [comment]
Thank you all @potiuk @jedcunningham 
My concern was not the rebase itself, but all the tests having to run again (after another approval) ;)

**Comment by potiuk** [comment]
Right. I sometimes rebase few times an hour (but only when I fix something) rebasing without change is indeed not needed (mostly). Unless there is an important change in main that is.

**Pull request merged**